radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java

```diff
@@ -259,15 +258,25 @@ private void handleMessage(ByteBuf buf) throws IOException {
   @Override
   public void channelActive(ChannelHandlerContext ctx) {
     // if we weren't able to determine peer's address earlier, it should be available now
-    if (this.remoteAddress.isEmpty()) {
```

Comment:
This is a bit unintuitive, but the check was there for a reason. If the node runs behind a reverse proxy (nginx), it gets peer's real ip address through a so called "proxy protocol", which is then supplied in a constructor.
Without the if statement, this would override that address with the address of Nginx.

Reply:
this could probably be refactored to make it more visible

Reply:
Good catch. Fixed and made visible.